contrib/automation_tests/core/orbit_e2e.py

```diff
@@ -51,52 +52,74 @@ def expect_true(self, cond, description):
     def expect_eq(self, left, right, description):
         self.expect_true(left == right, description)
 
+    def find_control(self, control_type=None, name=None, parent: BaseWrapper=None, name_contains=None,
+                     auto_id_leaf=None, qt_class=None, recurse=True, raise_on_failure=True) -> BaseWrapper:
+        """
+        Returns the first child of BaseWrapper that matches all of the search parameters.
+        As soon as a matching child is encountered, this function returns.
+
+        If no child is found, an exception is thrown.
+        """
+        if not parent:
+            parent = self.suite.top_window()
+        return find_control(parent, control_type=control_type, name=name, name_contains=name_contains,
+                            auto_id_leaf=auto_id_leaf, qt_class=qt_class, recurse=recurse,
+                            raise_on_failure=raise_on_failure)
+
+    def find_context_menu_item(self, text: str, raise_on_failure=True):
+        """
+        Specialized version of find_control for context menu items. This is provided due to the special
+        behavior of context menus: They are not parented underneath the current main window, but are treated
+        as a separate window.
+        """
+        return self.find_control('MenuItem', text, parent=self.suite.application.top_window(),
+                                 raise_on_failure=raise_on_failure)
+
     def _execute(self, **kwargs):
         """
         Provide this method in your fragment
         """
         pass
 
 
+class CloseOrbit(E2ETestCase):
+    def _execute(self):
+        # For some reason the line below does NOT work when Orbit is maximized - this is actually consistent with
+        # the results of AccessibilityInsights as the close button seems to have no on-screen rect...
+        # self.find_control("Button", "Close").click_input()
+        # ... so just send Alt + F4
+        send_keys('%{F4}')
+        logging.info('Closed Orbit.')
+
+
 class E2ETestSuite:
-    def __init__(self, test_name: str, tests: Iterable[E2ETestCase], dev_mode: bool = False,
-                 auto_connect: bool = True):
-        try:
-            self._application = Application(backend='uia').connect(title_re='orbitprofiler')
-        except Exception:
-            logging.error("Could not find Orbit application. Make sure to start Orbit before running E2E tests.")
-            raise
+    def __init__(self, test_name: str, tests: Iterable[E2ETestCase], dev_mode: bool = False):
```

Comment:
I had troubles getting the concepts of TestSuite, TestCase right at first. I'd suggest calling the param 'test_cases' (here and in files were the suites are implemented)

`def __init__(self, test_name: str, test_cases: Iterable[E2ETestCase], dev_mode: bool = False):`

Also I would rename the folder "fragments" to "test_cases" (since fragments are not a thing anymore, right?)

Even if this is something you agree on it is probably best to do this in a follow up PR.

Reply:
True, I was thinking about the same and then worried to make this too hard to review with those changes. I'll do that in a follow-up PR.

Reply:
sgtm!